src/renderer/layouts/default-layout.tsx

```diff
@@ -52,7 +54,26 @@ export const DefaultLayout = ({ shell }: DefaultLayoutProps) => {
   const { windowBarStyle } = useWindowSettings();
   const { opened, ...handlers } = useCommandPalette();
   const { bindings } = useHotkeySettings();
+  const localSettings = isElectron() ? window.electron.localSettings : null;
+  const settings = useGeneralSettings();
+  const { setSettings } = useSettingsStoreActions();
 
+  const updateZoom = (increase: number) => {
+    const newVal = settings.zoomFactor + increase;
+    if (newVal > 300 || newVal < 50 || !isElectron()) return;
+    setSettings({
+      general: {
+        ...settings,
+        zoomFactor: newVal,
+      },
+    });
+    console.log(settings.zoomFactor);
+    localSettings?.setZoomFactor(settings.zoomFactor);
+  };
+  localSettings?.setZoomFactor(settings.zoomFactor);
+
+  useHotkeys([[bindings.zoomIn.hotkey, () => updateZoom(5)]]);
+  useHotkeys([[bindings.zoomOut.hotkey, () => updateZoom(-5)]]);
   useHotkeys([[bindings.globalSearch.hotkey, () => handlers.open()]]);
```

Comment:
You can combine this into a single useHotkeys hook instead of instantiating it three times. 
- [Example from docs](https://mantine.dev/hooks/use-hotkeys/)

Also, since this is a desktop-only feature you should probably conditionally add these hotkeys.

```typescript
const zoomHotkeys: HotkeyItem[] = [
  [bindings.zoomIn.hotkey, () => updateZoom(5)],
  [bindings.zoomOut.hotkey, () => updateZoom(-5)],
];

useHotkeys([
  [bindings.globalSearch.hotkey, () => handlers.open()],
  ...(isElectron() ? zoomHotkeys : []),
]);
```